src/kayenta/utils/duration.spec.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  defaultDurationObject,
+  defaultDurationString,
+  getDurationString,
+  parseDurationString,
+  IDuration,
+} from './duration';
+
+describe('Duration utils', () => {
+  let durationStrings: string[];
+  let invalidDurationStrings: any[];
+  let durationObjects: IDuration[];
+  let invalidDurationObjects: any[];
+
+  beforeEach(() => {
+    durationStrings = ['PT1H0M', 'PT0H10M', 'PT0H0M', 'PT20H6M'];
+
+    invalidDurationStrings = ['invalid', { k: 'not_a_string' }, null];
+
+    durationObjects = [
+      { hours: 1, minutes: 0 },
+      { hours: 0, minutes: 10 },
+      { hours: 0, minutes: 0 },
+      { hours: 20, minutes: 6 },
+    ];
+
+    invalidDurationObjects = [null, 'not_an_object', { hours: 'not_an_int', minutes: -1 }];
+  });
+
+  describe('parseDurationString', () => {
+    it('converts duration string to object', () => {
+      durationStrings.forEach((durationString, i) => {
+        expect(parseDurationString(durationString)).toEqual(durationObjects[i]);
```

Comment:
This might be a little easier to read this if you were iterating through a list that had expected `inputs` and `outputs`:
```
const tests = [
  {
     input: 'PT1H0M',
     output: { hours: 1, minutes: 0 },
  },
  ...
]
```
Then you could just say 
```
tests.forEach(({ input, output }) => expect(parseDurationString(input)).toEqual(output))
```

Reply:
Great idea, updated!